### PR TITLE
chore: migrate js/ts project

### PR DESCRIPTION
**Pull request by blackchoey**
migrate webapp bot, function bot and function api projects

### CODE REVIEW

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml [hunk 1]
```diff
@@ -28,19 +35,82 @@ deploy:
       workingDirectory: tabs
```
Comment (Siglud) [inline packages/fx-core/templates/core/v3Migration/js.ts.app.yml:35]: enable static website need to be like:
```
  - uses: azureStorage/enableStaticWebsite
    with:
      storageResourceId: ${{TAB_RESOURCE_ID}}
      indexPage: index.html
      errorPage: error.html
```

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml [hunk 2]
```diff
@@ -28,19 +35,82 @@ deploy:
...
+    with:
+      args: install
+    {{#if isTypescript}}
+  - uses: cli/runNpmCommand
+    with:
+      args: run build --if-present
+    {{/if}}
+  - uses: azureFunctions/deploy
+    with:
+      distributionPath: ./ # Deploy base folder
```
Comment (swatDong) [inline packages/fx-core/templates/core/v3Migration/js.ts.app.yml:69]: Any difference between `.` and `./`? I see different values in this file...

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml [hunk 3]
```diff
@@ -28,19 +35,82 @@ deploy:
...
+    with:
+      workingDirectory: bot
+      args: install
+    {{#if isTypescript}}
+  - uses: cli/runNpmCommand # Run npm command
+    with:
+      workingDirectory: bot
+      args: run build --if-present
+    {{/if}}
+  - uses: azureAppService/deploy # Deploy bits to Azure Storage Static Website
```
Comment (swatDong) [inline packages/fx-core/templates/core/v3Migration/js.ts.app.yml:84]: Deploy bits to *Azure App Serivce*

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml [hunk 4]
```diff
@@ -28,19 +35,82 @@ deploy:
...
+    with:
+      workingDirectory: bot
+      distributionPath: . # Deploy base folder. 
+      ignoreFile: # Can be changed to any ignore file location, leave blank will ignore nothing
```
Comment (Siglud) [inline packages/fx-core/templates/core/v3Migration/js.ts.app.yml:88]: Deploy to Azure App Service with a nodeJS project requires at least one ignore file that ignores the .deployment folder

**Review by xzf0587 — APPROVED** [review]

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml (hunk 2 above)
Comment (blackchoey) [reply packages/fx-core/templates/core/v3Migration/js.ts.app.yml:69]: I think there's no difference and the scaffolding templates are mixing them.
I changed most of them to `.` in my template but missed this, will update my PR.

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml (hunk 3 above)
Comment (blackchoey) [reply packages/fx-core/templates/core/v3Migration/js.ts.app.yml:84]: thanks, updated

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml (hunk 1 above)
Comment (blackchoey) [reply packages/fx-core/templates/core/v3Migration/js.ts.app.yml:35]: This file is a template, the `$\{{ {{~placeholderMappings.[state.fx-resource-frontend-hosting.storageResourceId]~}} }}` will be replaced with real resource id's environment name based on user's bicep content.

**Review by dooriya — DISMISSED** [review]

**Review by hund030 — DISMISSED** [review]

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml (hunk 4 above)
Comment (blackchoey) [reply packages/fx-core/templates/core/v3Migration/js.ts.app.yml:88]: thanks, let me track this separately

**Review by swatDong — APPROVED** [review]

**Review by xzf0587 — APPROVED** [review]

**Review by dooriya — APPROVED** [review]

**Pull request merged**